RecoPixelVertexing/PixelTriplets/src/CACut.h

```diff
@@ -0,0 +1,114 @@
+#ifndef RecoPixelVertexing_PixelTriplets_src_CACut_h
+#define RecoPixelVertexing_PixelTriplets_src_CACut_h
+
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "CAGraph.h"
+
+class CACut {
+public:
+  explicit CACut(const double defaultCut, const edm::ParameterSet &tripletCuts) {
+    setCutValuesByTripletNames(defaultCut, tripletCuts);
+  }
+
+  void setCutValuesByTripletNames(double defaultCut, const edm::ParameterSet &tripletCuts) {
+    std::vector<std::string> tripletNames = tripletCuts.getParameterNames();
+
+    for (const std::string &thisTripletName : tripletNames) {
+      CAValueByTripletName thisTriplet;
+      thisTriplet.tripletName = thisTripletName;
+
+      float thisCutValue = tripletCuts.getParameter<double>(thisTripletName);
+      if (thisCutValue > 0) {
+        thisTriplet.cutValue = thisCutValue;
+      } else {
+        //TODO: Uncomment the following line once the tuned values are added to the PSet
+        //edm::LogWarning("Configuration") << "Layer triplet '" << tripletName <<"' not in the CACuts parameter set. Using default cut value: " << defaultCut;
+        thisTriplet.cutValue = defaultCut;
+      }
+
+      valuesByTripletNames_.emplace_back(thisTriplet);
+    }
+  }
+
+  void setCutValuesByLayerIds(CAGraph &caLayers) {
+    for (const auto &thisTriplet : valuesByTripletNames_) {
+      CAValueByLayerIds thisCACut;
+
+      // Triplet name, e.g. "layerA__layerB__layerC"
+      std::string layersToSet = thisTriplet.tripletName;
+
+      // Layer names and id's
+      std::string layerName;
+      std::size_t layerPos = 0;
```

Comment:
There seems to be no use of `layerName` or `layerPos` outside of the following loop, so the declarations would be better moved inside the loop as well.